.github/workflows/e2e-tests.yml

```diff
@@ -1,11 +1,15 @@
-name: E2E Tests for PR
+name: E2E Tests
 
 on:
+  push:
+    branches:
+      - "master"
+      - 'release-**'
   pull_request:
     types: [opened, synchronize, reopened, ready_for_review]
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
+  group: ${{ github.workflow }}-${{ github.head_ref && github.ref || github.run_id }}
   cancel-in-progress: true
```

Comment:
We now have to apply the conditional concurrency that's proven to work.
It will distinguish between pushes to master (will not cancel anything) and `synchronize` event (pushes within one's PR).